src/common/walFilter/versions/recordProcessGPDB7.c

```diff
@@ -0,0 +1,436 @@
+                    return (const RelFileNode *) mainData;
+
+                case XLOG_HEAP2_NEW_CID:
+                {
+                    const xl_heap_new_cid *const xlrec = mainData;
+                    return &xlrec->target;
+                }
+
+                default:
+                    break;
+            }
+            break;
+
+        case RM7_BTREE_ID:
+            if (info == XLOG_BTREE_REUSE_PAGE)
+                return (const RelFileNode *) mainData;
+            break;
+
+        case RM7_GIN_ID:
+            switch (info)
+            {
+                case XLOG_GIN_SPLIT:
+                case XLOG_GIN_UPDATE_META_PAGE:
+                    return (const RelFileNode *) mainData;
+
+                default:
+                    break;
+            }
+            break;
+
+        case RM7_GIST_ID:
+            if (info == XLOG_GIST_PAGE_REUSE)
+                return (const RelFileNode *) mainData;
+            break;
+
+        case RM7_SEQ_ID:
+            if (info == XLOG_SEQ_LOG)
+                return (const RelFileNode *) mainData;
+            THROW_FMT(FormatError, "unknown Sequence: %d", info);
+
+        case RM7_BITMAP_ID:
+            switch (info)
+            {
+                case XLOG_BITMAP_INSERT_WORDS:
+                case XLOG_BITMAP_UPDATEWORD:
+                case XLOG_BITMAP_UPDATEWORDS:
+                case XLOG_BITMAP_INSERT_LOVITEM:
+                case XLOG_BITMAP_INSERT_BITMAP_LASTWORDS:
+                case XLOG_BITMAP_INSERT_META:
+                    return (const RelFileNode *) mainData;
+
+                default:
+                    break;
+            }
+            break;
+
+        case RM7_APPEND_ONLY_ID:
+            switch (info)
+            {
+                case XLOG_APPENDONLY_INSERT:
+                case XLOG_APPENDONLY_TRUNCATE:
+                    return (const RelFileNode *) mainData;
+
+                default:
+                    THROW_FMT(FormatError, "unknown Appendonly: %d", info);
+            }
+    }
+
+    return NULL;
+}
+
+static List *
+getRelFileNodes(XLogRecordGPDB7 *const record, PgPageSize pageSize)
+{
+    #define COPY_HEADER_FIELD(_dst, _size) \
+    do                                     \
+    {                                      \
+        if (remaining < _size)             \
+            return NULL;                   \
+        memcpy(_dst, ptr, _size);          \
+        ptr += _size;                      \
+        remaining -= (uint32_t) _size;     \
+    }                                      \
+    while(0)
+
+    FUNCTION_LOG_BEGIN(logLevelTrace);
+        FUNCTION_LOG_PARAM(XLOG_RECORD_GPDB7, record);
+        FUNCTION_LOG_PARAM(INT, (int) pageSize);
+    FUNCTION_LOG_END();
+
+    uint32_t remaining = record->xl_tot_len - (uint32_t) sizeof(XLogRecordGPDB7);
+    /* Decode the headers */
+    uint32_t datatotal = 0;
+    int maxBlockId = -1;
+    char *ptr = (char *) record;
+    ptr += sizeof(XLogRecordGPDB7);
```

Comment:
Let's cast to char data after the header, but not the header
```suggestion
    char *ptr = (char *) (record + 1);
```

Reply:
Are you sure it's more readable?

Reply:
As for me, yes

Reply:
I don't think that's more readable either. Maybe leave it as it is, or give `*ptr` a more descriptive name?

But the definition can be shortened (up to the author):
```suggestion
    char *ptr = (char *) record + sizeof(XLogRecordGPDB7);
```